extensions/ql-vscode/test/pure-tests/helpers-pure.test.ts

```diff
@@ -19,4 +19,38 @@ describe('helpers-pure', () => {
       expect(e.message).to.eq('opps');
     }
   });
+
+  describe('error message shortener', () => {
+    const WINDOWS_LINE_END = '\r\n';
+    const LINUX_LINE_END = '\n';
+
+    function runTest(input: string[], expected: string[]) {
+      expect(shortenErrorMessage(input.join(WINDOWS_LINE_END)).split(WINDOWS_LINE_END)).eql(expected);
+      expect(shortenErrorMessage(input.join(LINUX_LINE_END)).split(LINUX_LINE_END)).eql(expected);
+    }
+
+    it('removes stack traces', async () => {
+      const input = ['first line', '    com.a.java.class.command(Class.java:65)', 'last line'];
+      const expected = ['first line', 'last line'];
+      runTest(input, expected);
+    });
+
+    it('removes log lines lines', async () => {
+      const input = ['first line', '[2020-02-15 09:10:15] Some logging information', 'last line'];
+      const expected = ['first line', 'last line'];
+      runTest(input, expected);
+    });
+
+    it('strips whitespace', async () => {
+      const input = ['    first line', '    last line'];
+      const expected = ['first line', 'last line'];
+      runTest(input, expected);
+    });
+
+    it('removes duplicate lines', async () => {
+      const input = ['first line', 'last line', 'last line'];
```

Comment:
What's the behaviour here:
```suggestion
      const input = ['first line', 'last line', 'last line', 'last line'];
```

Looking at the regex, I'm not sure. Could you add a unit test for that?